src/cmd/gz.cc

```diff
@@ -37,47 +40,119 @@
 #include "gz/sim/Server.hh"
 #include "gz/sim/ServerConfig.hh"
 
+#include "gz/sim/Util.hh"
+
+#ifdef WITH_GUI
 #include "gz/sim/gui/Gui.hh"
+#endif
 
 using namespace gz;
 
 //////////////////////////////////////////////////
-extern "C" char *gzSimVersion()
+void cmdVerbosity(const int _verbosity)
 {
-  return strdup(GZ_SIM_VERSION_FULL);
+  common::Console::SetVerbosity(_verbosity);
+
+  // SDFormat only has 2 levels: quiet / loud. Let sim users suppress all SDF
+  // console output with zero verbosity.
+  if (_verbosity == 0)
+  {
+    sdf::Console::Instance()->SetQuiet(true);
+  }
 }
 
 //////////////////////////////////////////////////
-extern "C" char *simVersionHeader()
+const std::string worldInstallDir()
 {
-  return strdup(GZ_SIM_VERSION_HEADER);
+  static std::string worldInstallDir = gz::sim::getWorldInstallDir();
+  return worldInstallDir;
 }
 
 //////////////////////////////////////////////////
-extern "C" void cmdVerbosity(
-    const char *_verbosity)
+int checkFile(std::string &_file)
 {
-  int verbosity = std::atoi(_verbosity);
-  common::Console::SetVerbosity(verbosity);
-
-  // SDFormat only has 2 levels: quiet / loud. Let sim users suppress all SDF
-  // console output with zero verbosity.
-  if (verbosity == 0)
+  // Check if passed string is valid
+  if(!_file.empty())
```

Comment:
should there be an `else` branch that returns `-1` if `_file` is empty?

Reply:
c5c40b4aa

Reply:
the whole thing did not work since the logic behind this and your next comment was to run checksdf and parsesdf functions even if the input file was empty and return ok if that case. I changed the logic not to pass the files in [f8d454e](https://github.com/gazebosim/gz-sim/pull/2849/commits/f8d454e3ae942bb7121277b3dd04ec6d29bf3b3f). There are some problems in that code (gz-main-gui should not accept  sdf inputs) but I prefer not to change this PR more and create new ones. The problem is registered in the PR description.